docs/sources/tempo/troubleshooting/long-running-traces.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Long-running traces
+description: Troubleshoot search results when using long-running traces
+weight: 479
+aliases:
+  - ../operations/troubleshooting/long-running-traces/
+---
+
+# Long-running traces
+
+An issue arises in Tempo when a user exercises a usage pattern referred to as
+long-running traces. This happens when Tempo receives spans for a trace,
+followed by a delay, and then Tempo receives additional spans for the same
+trace. If the delay between spans is great enough, the spans end up in
+different blocks, which can lead to inconsistency in a few ways.
+
+1. When using TraceQL search, the duration information only pertains to a
+   subset of the blocks that contain a trace. This happens because Tempo
+   consults only enough blocks to know the TraceID of the matching spans. When
+   performing a TraceID lookup, Tempo searches for all parts of a trace in all
+   matching blocks, which yields greater accuracy when combined.
+
+1. When using structural operators, the conditions may match on different
+   blocks, and so results can be confusing.
+
+Users can tune the `ingester.trace_idle_period` configuration to allow for
+greater control about when traces are written to a block. Extending this beyond
+the default `10s` can allow for long running trace to be co-located in the same
+block, but take into account other considerations around memory consumption on
+the ingesters. Currently this setting isn't per-tenant, and so adjusting
+affects all ingester instances.
+
+Tempo publishes a `tempo_warnings_total` metric from several components, which
+can aid in understanding when this situation arises. In particular, the following query can be used to know what percentage of traces which are flushed to the wall are connected.
+
+```
+1 - sum(rate(tempo_warnings_total{reason="disconnected_trace_flushed_to_wal"}[5m])) / sum(rate(tempo_ingester_traces_created_total{}[5m]))
+```
+
+Users of this pattern may also be interested in the
+`rootless_trace_flushed_to_wal` reason to know when a trace is flushed to the
+wall without a root trace.
```

Comment:
Is this correct? Is "this pattern" referring to long-running traces or the preceding query example? 
```suggestion
If you have long-running traces, you may also be interested in the
`rootless_trace_flushed_to_wal` reason to know when a trace is flushed to the
wall without a root trace.
```